src/main/kotlin/org/rust/ide/intentions/IfLetToMatchIntention.kt

```diff
@@ -66,6 +73,23 @@ class IfLetToMatchIntention : RsElementBaseIntentionAction<IfLetToMatchIntention
         ifStmt.replace(matchExpression)
     }
 
+    private fun isExhaustiveOptionOrResult(arms: List<MatchArm>): Boolean = arms.all {
+        it.matchArm.isIrrefutable
+    } && arms.map { it.matchArm.text.substringBefore('(') }.let {
+        "Some" in it && "None" in it || "Ok" in it && "Err" in it
+    }
+
+    private fun missingBranch(arms: List<MatchArm>): String = if (arms.size == 1) {
+        val pat = arms.first().matchArm
+        if (pat.isIrrefutable) when (pat.text.substringBefore('(')) {
```

Comment:
This is not enough to make sure that we are dealing with items from the core library. You can see how this is implemented in `UnwrapToMatchIntention`. You can (maybe) reuse some code from there.

And let's rewrite this code a little bit:
```
private fun missingBranch(arms: List<MatchArm>): String {
    if (arms.size <= 1) return "_"
    val pat = arms.first().matchArm
    if (!pat.isIrrefutable) return "_"
    ...
```

Reply:
Done.